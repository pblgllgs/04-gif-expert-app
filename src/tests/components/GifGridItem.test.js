import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('Pruebas de GifGridItem', () => {

    const title = 'Titulo';
    const url = 'http://localhost:3000/uploads/foto.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test("Debe mostrar <GifGridItem /> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el title', () => {
        const parrafo = wrapper.find("p").text();
        expect(parrafo.trim()).toBe(title);
    });

    test('Debe de tener un img con src y alt', () => {
        const imagen = wrapper.find("img").props();
        expect(imagen.src).toBe(url);
        expect(imagen.alt).toBe(title);
    });

    test('Debe de tener un div con clasename animate__fadeIn', () => {
        const div = wrapper.find("div").props();
        expect(div.className).toContain('animate__fadeIn');
    });

    test('Debe de tener un div con clasename animate__fadeIn 2', () => {
        const div = wrapper.find("div").props();
        const className = div.className;
        const arr = className.split(' ');
        expect(arr).toContain('animate__fadeIn')
    });
});
