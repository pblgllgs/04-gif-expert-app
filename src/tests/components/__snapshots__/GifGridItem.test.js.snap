// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGridItem Debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="http://localhost:3000/uploads/foto.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
